config/env.config.js

```diff
@@ -73,34 +79,47 @@ const RedirectPlugin = function(options) {
+                        var targetUrl = (getParameterByName("targetEnvironment") || "").toLowerCase();
+
+                        // Set target environment for origin environment to redirect to
+                        if (targetUrl.length > 0) {
+                            targetUrl = decodeURIComponent(targetUrl)
+                            // Clear origin environment's local storage if target = origin
+                            if (window.location.href.toLowerCase().indexOf(targetUrl) != -1) {
+                                window.localStorage.removeItem("${PLAYGROUND_REDIRECT}");
+                                return;
                             }
+
+                            window.localStorage.setItem("${PLAYGROUND_REDIRECT}", targetUrl);
                         }
 
-                        if (window.localStorage.getItem("environmentConfig")) {
-                            var originParam = "" + 
-                            (window.location.search ? "&" : "?") + 
-                            "originEnvironment=" + 
-                            encodeURIComponent(window.location.origin);
+                        // Redirect origin environment to target
+                        if (window.localStorage.getItem("${PLAYGROUND_REDIRECT}")) {
+                            var originParam = [
+                                (window.location.search ? "&" : "?"), 
+                                "originEnvironment=",
+                                encodeURIComponent(window.location.origin)
+                            ].join("");
 
-                            window.location.href = window.localStorage.getItem("environmentConfig") +
-                            window.location.pathname +
-                            window.location.search + 
-                            originParam + 
-                            window.location.hash;
+                            window.location.href = [
+                                window.localStorage.getItem("${PLAYGROUND_REDIRECT}"),
+                                window.location.pathname,
+                                window.location.search,
+                                originParam,
+                                window.location.hash
+                            ].join("");
                         }
 
-                        if (originUrl && !window.localStorage.getItem("originEnvironment")) {
-                            window.localStorage.setItem("originEnvironment", decodeURIComponent(originUrl));
+                        // Point app environment back to origin if user is not in origin
+                        if (originUrl.length > 0) {
+                            window.localStorage.setItem("${PLAYGROUND_ORIGIN}", decodeURIComponent(originUrl).toLowerCase());
                         }
-                    </script>` + 
+
+                        // Environment is already configured
```

Comment:
Nit, I would add a "**If reached here**, environment ...".  That way it's clear it's a complete sentence, and not a comment that got left accidentally from some refactoring.